x-pack/plugins/enterprise_search/public/applications/app_search/components/credentials/credentials_logic.test.ts

```diff
@@ -0,0 +1,1006 @@
+    describe('activeApiToken', () => {
+      it('should set the value of access_all_engines and clear out engines list if true', () => {
+        mount({
+          activeApiToken: {
+            ...newToken,
+            access_all_engines: false,
+            engines: ['someEngine', 'anotherEngine'],
+          },
+        });
+        CredentialsLogic.actions.setAccessAllEngines(true);
+        expect(CredentialsLogic.values.activeApiToken).toEqual({
+          ...newToken,
+          engines: [],
+          access_all_engines: true,
+        });
+      });
+    });
+
+    it('should set the value of access_all_engines and but maintain engines list if false', () => {
+      mount({
+        activeApiToken: {
+          ...newToken,
+          access_all_engines: true,
+          engines: ['someEngine', 'anotherEngine'],
+        },
+      });
+      CredentialsLogic.actions.setAccessAllEngines(false);
+      expect(CredentialsLogic.values.activeApiToken).toEqual({
+        ...newToken,
+        access_all_engines: false,
+        engines: ['someEngine', 'anotherEngine'],
+      });
+    });
+
+    it('should not change any other values', () => {
+      mount();
+      CredentialsLogic.actions.setAccessAllEngines(true);
+      expect(CredentialsLogic.values).toEqual({
+        ...DEFAULT_VALUES,
+        activeApiToken: expect.any(Object),
+      });
+    });
+  });
+
+  describe('onApiKeyDelete', () => {
+    describe('apiTokens', () => {
+      it('should remove specified token from apiTokens if name matches', () => {
+        mount({
+          apiTokens: [newToken],
+        });
+
+        CredentialsLogic.actions.onApiKeyDelete(newToken.name);
+        expect(CredentialsLogic.values.apiTokens).toEqual([]);
+      });
+
+      it('should not remove specified token from apiTokens if name does not match', () => {
+        mount({
+          apiTokens: [newToken],
+        });
+
+        CredentialsLogic.actions.onApiKeyDelete('foo');
+        expect(CredentialsLogic.values.apiTokens).toEqual([newToken]);
+      });
+    });
+
+    it('should not change any other values', () => {
+      mount();
+      CredentialsLogic.actions.onApiKeyDelete('foo');
+      expect(CredentialsLogic.values).toEqual({
+        ...DEFAULT_VALUES,
+        apiTokens: expect.any(Array),
+      });
+    });
+  });
+
+  describe('onApiTokenCreateSuccess', () => {
+    describe('apiTokens', () => {
+      const existingToken = {
+        name: 'some_token',
+        type: PRIVATE,
+      };
+
+      it('should add the provided token to the apiTokens list', () => {
+        mount({
+          apiTokens: [existingToken],
+        });
+
+        CredentialsLogic.actions.onApiTokenCreateSuccess(newToken);
+        expect(CredentialsLogic.values.apiTokens).toEqual([existingToken, newToken]);
+      });
+    });
+
+    describe('activeApiToken', () => {
+      // TODO It is weird that methods like this update activeApiToken but not activeApiTokenIsExisting...
```

Comment:
I found some oddities around this logic file that I just noted in TODOs for now. This is a good example. Everywhere else in the code, when we change the activeApiToken, we also update `activeApiTokenIsExisting`. They are manually kept in sync. Here, we do not update `activeApiTokenIsExisting`.

I think the ultimate solution is to change `activeApiTokenIsExisting` to be a selector, so we don't need to try to keep it in sync everywhere.

Reply:
I think this is a super good example of unit tests catching a design flaw in code! I'd be +++ for fixing this behavior - this does indeed sound like the exact use-case for a selector.

Also while we're here, would we have any objections to renaming `activeApiTokenIsExisting`? It's fairly awkward-sounding - maybe `doesActiveApiTokenExist` or even just `hasActiveApiToken` instead?

Reply:
BTW, would `activeTokenRawName` be another good use-case to convert to a selector?

Reply:
Since we know this works as is, and this PR is already big, I'd like to address this is a follow-up PR.

Reply:
Sounds great! No rush at all!